client/src/pages/Contest/Contest.tsx

```diff
@@ -48,21 +49,33 @@ export default function Contest(): JSX.Element {
       imageFiles: string[];
     }>,
   ) => {
-    createContest(title, description, prizeAmount, createDeadlineDate(date, time, timeZone), imageFiles).then(
-      (data) => {
-        if (data.error) {
-          setSubmitting(false);
-          updateSnackBarMessage(data.error.message);
-        } else if (data.success) {
-          history.push('/dashboard');
-          updateSnackBarMessage('Created Contest Successfully');
-        } else {
-          console.error({ data });
-          setSubmitting(false);
-          updateSnackBarMessage('An unexpected error occured. Please try again');
-        }
-      },
+    const contest = await createContest(
+      title,
+      description,
+      prizeAmount,
+      createDeadlineDate(date, time, timeZone),
+      imageFiles,
     );
+
+    if (contest.error) {
+      setSubmitting(false);
+      updateSnackBarMessage(contest.error.message);
+    } else if (contest.success) {
+      const stripeUser = await stripeCustomer();
+
+      if (stripeUser.existingStripeCustomer) {
+        setSubmitting(false);
+        updateSnackBarMessage('Card on file will be charged when you select the winner!');
+        history.push('/dashboard');
+      } else {
+        setSubmitting(false);
+        history.push('/payment-details');
+      }
+    } else {
```

Comment:
What is the reason for this else clause as we are already checking success/failure in the above two blocks?

Reply:
it came from the previous contest integration PR. I will remove it.

Reply:
successfully implemented in commit 8340f128b930c04eb81e229095baa2623bda7942